components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt

```diff
@@ -156,7 +156,17 @@ class GeckoEngineView @JvmOverloads constructor(
      * GeckoEngineSession) - since the previous one is no longer usable.
      */
     private fun rebind() {
-        currentSession?.let { currentGeckoView.setSession(it.geckoSession) }
+        try {
+            currentSession?.let { currentGeckoView.setSession(it.geckoSession) }
+        } catch (e: IllegalStateException) {
+            // This is to debug "Display not attached" crashes
+            val otherActivityClassName =
+                currentSession?.geckoSession?.accessibility?.view?.context?.javaClass?.simpleName
+            val activityClassName = context.javaClass.simpleName
+            val msg =
+                "SET SESSION: Current activity: $activityClassName Other activity: $otherActivityClassName"
```

Comment:
We just recently added the hashcodes when logging the activities. I think this is helpful here too so we can tell if it's the same instance or not: https://github.com/mozilla-mobile/android-components/blob/master/components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineView.kt#L46

Reply:
Done! Thanks.

Reply:
Thanks!